### DISCUSSION TITLE
Frequent errors when changing mode in latex files

### DISCUSSION THREAD

**Discussion by a-torgovitsky** [LazyVim/LazyVim#4142] [Q&A]
I'm getting the following error popping up often when writing in latex:

```
   Error  10:27:09 AM msg_show.lua_error E5108: Error executing lua Keyboard interrupt
stack traceback:
	[C]: in function 'has'
	...t/.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/ui.lua:15: in function 'get_signs'
	...t/.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/ui.lua:106: in function <...t/.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/ui.lua:94>
```

Any idea how to debug this? I'm not quite sure what causes it, but it seems like it happens sometimes when I am moving from insert to normal mode. Not always though.

**Comment by folke** [comment]
You're doing `ctrl-c`? So yes, that would yield errors all over Neovim

**Reply by a-torgovitsky** [reply]
Yes that's it, thanks. Is remapping `<C-c>` to `<Esc>` a reasonable thing to do?

**Reply by folke** [reply]
You can remap it to anything. I personally remapped it to `norm! ciw`

**Reply by a-torgovitsky** [reply]
Ok, I'll try remapping `<C-c>` to `<Esc>`. But if you don't mind I'm also curious as to how you usually go from insert to normal? Do you like `<Esc>`?

**Reply by dotfrag** [reply]
Not sure if this helps but I have my caps-lock mapped to esc. In fact, I'm so used to it that I can't use a computer without this mapping.

**Reply by a-torgovitsky** [reply]
Thanks that sounds like a great choice, although I'm in the same position as you with `<C-c>` by now having built-in many years of muscle memory. Still I'm interested in what others use for this, as `<Esc>` seems inconvenient.

**Reply by folke** [reply]
I have capslock mapped to `<esc>` on single press and `<ctrl>` on holding.

**Comment by a-torgovitsky** [comment]
Hmm, so I actually started having this error come up again, even after I made sure to use `<Esc>` instead of `<C-c>`. The stacktrace looks a bit longer now, but the first two lines are the same:

```
E5108: Error executing lua Keyboard interrupt
stack traceback:
	[C]: in function 'has'
	...t/.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/ui.lua:15: in function 'get_signs'
	...t/.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/ui.lua:106: in function <...t/.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/ui.lua:94>
	[C]: in function 'nvim__redraw'
	...ocal/share/nvim/lazy/noice.nvim/lua/noice/util/hacks.lua:38: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```
Any ideas?

**Reply by a-torgovitsky** [reply]
The error disappears if I disable noice.
I wonder if it might be related to [this discussion](https://github.com/lervag/vimtex/issues/2661) about another unexpected interaction between noice and vimtex?
I also have that problem, but it too disappears if I disable noice.

**Reply by dpetka2001** [reply]
Either create an issue at noice github repo and hope that the maintainer will give attention to it (although from the issue at vimtex that you linked this doesn't seem likely, but that's just me reading an over a year old comment and it might not be this way now) or disable noice if it causes you trouble. No need for you to use software that breaks your workflow and just the same holds true for the developers that probably won't invest their time for every niche case if it works for 99% of the time.

**Reply by a-torgovitsky** [reply]
Yeah that's what I've done, or more specifically I just disable noice for .tex files, because I haven't experienced any problems with it in other settings.
